### PR TITLE
copy ApiDoc folder from docs repo

### PR DESCRIPTION
**Pull request by phryneas**
This PR moves over Components from https://github.com/apollographql/docs/pull/646 into this repository so we can iterate on them faster over here, without having to do two PRs every time.

For a better diff, take a look at https://github.com/apollographql/apollo-client/compare/391be6a8f9f473dc6aaac284615dabfc28e5fe7a..pr/move-ApiDocs (skips the first commit which only copies files)

### Checklist:

- [ ] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@
   npm run docmodel
   cd ../
   rm -rf monodocs
-  git clone https://github.com/apollographql/docs --branch main --single-branch monodocs
+  git clone https://github.com/apollographql/docs --branch pr/expose-hook --single-branch monodocs
```
[the remark this answers is not in the document]
Comment (phryneas) [reply netlify.toml:16]: It has to stay until we have merged it over in `docs`, which can only be merged in after this PR 😅 
So I'll merge this, then `docs`, then push a correction to `main` ^^

File: docs/shared/ApiDoc/DocBlock.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  example = false,
+  since = false,
+  deprecated = false,
+  collapsible = false
+}) {
+  const getItem = useApiDocContext();
+  const item = getItem(canonicalReference);
+  let jsx = null;
+
+  switch (true) {
```
[the remark this answers is not in the document]
Comment (phryneas) [reply docs/shared/ApiDoc/DocBlock.js]: https://github.com/apollographql/apollo-client/pull/11416/commits/fdd3889a6803453f2ff96b93834b666c93e2bb80

File: docs/shared/ApiDoc/ResponsiveGrid.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export function ResponsiveGrid({children}) {
+  {
+    /*
+        responsiveness not regarding screen width, but actual available space:
+        if less than 350px, show only one column
+        show at most two columns (that's where the 45% hack comes - 45% * 3 won't fit)
+       */
+  }
```
[the remark this answers is not in the document]
Comment (phryneas) [reply docs/shared/ApiDoc/ResponsiveGrid.js]: https://github.com/apollographql/apollo-client/pull/11416/commits/c72edb4dcd3176ade76b26038b5539382506b309

File: docs/shared/ApiDoc/Heading.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  2: 'xl',
+  3: 'lg',
+  4: 'md',
+  5: 'sm',
+  6: 'xs'
+};
+
+export function ApiDocHeading({
+  canonicalReference,
+  headingLevel = 3,
```
[the remark this answers is not in the document]
Comment (phryneas) [reply docs/shared/ApiDoc/Heading.js]: https://github.com/apollographql/apollo-client/pull/11416/commits/727302fc20b52bedaad20bc42a878112d4c29e56

**Review by jerelmiller — APPROVED** [review]
Thanks for making those updates!

**Pull request merged**